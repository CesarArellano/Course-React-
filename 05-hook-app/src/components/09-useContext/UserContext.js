import { createContext } from 'react';

//Higher Order Component
export const UserContext = createContext(null);
