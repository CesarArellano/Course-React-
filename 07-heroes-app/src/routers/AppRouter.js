import React, { useContext } from 'react'
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
  const { user:{ logged } } = useContext(AuthContext);
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute 
            exact 
            path="/login" 
            isAuthenticated={ logged }
            component={ LoginScreen } 
          />
          <PrivateRoute 
            path="/" 
            isAuthenticated={ logged }
            component={ DashboardRoutes } 
          />
        </Switch>
      </div>
    </Router>
  );
}
