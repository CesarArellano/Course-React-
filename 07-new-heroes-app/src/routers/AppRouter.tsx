import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { PrivateRoute } from './PrivateRoute';
import { LoginScreen } from '../components/login/LoginScreen';
import { PublicRoute } from './PublicRoute';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/login" element={ 
          <PublicRoute>
            <LoginScreen />
          </PublicRoute>
          } 
        />
        <Route path="/*" element={ 
            <PrivateRoute>
              <DashboardRoutes />
            </PrivateRoute>
          }
        />
        {/* <Route path="/*" element={ <DashboardRoutes /> }/> */}
      </Routes>
    </BrowserRouter>
  );
}
