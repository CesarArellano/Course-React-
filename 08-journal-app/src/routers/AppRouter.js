import React, { useEffect, useState } from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { firebase } from '../firebase/firebase-config';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';

export const AppRouter = () => {

  const dispatch = useDispatch();
  const [checking, setChecking] = useState(true);
  const [ isLoggedIn, setIsLoggedIn ] = useState(false)
  
  useEffect(() => {
    firebase.auth().onAuthStateChanged( (user) => {
      if( user?.uid ) {
        dispatch(login(user.uid, user.displayName))
        setIsLoggedIn(true);
      } else {
        setIsLoggedIn(false);
      }
      setChecking(false);
    });
  }, [dispatch, setChecking]);

  if( checking ) {
    return (
      <div className="center-complete-box">
        <div className="utils__spinner"></div>
      </div>
    )
  }
  
  return (
    <Router>
      <Switch>
        <Route path="/auth" component={ AuthRouter } />
        <Route exact path="/" component={ JournalScreen } />
      </Switch>
    </Router>
  )
}
