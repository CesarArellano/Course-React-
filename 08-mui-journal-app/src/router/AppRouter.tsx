import { useSelector, useDispatch } from 'react-redux';
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom'
import { AuthRoutes } from '../auth/routes/AuthRoutes';
import { JournalRoutes } from '../journal/routes/JournalRoutes';
import { CheckingAuth } from '../ui';
import { useEffect } from 'react';
import { onAuthStateChanged } from 'firebase/auth';
import { FirebaseAuth } from '../firebase/config';
import { login, logout } from '../store/auth';

export const AppRouter = () => {
  const { status } = useSelector<any, any>(state => state.auth);
  const dispatch = useDispatch<any>();
  useEffect(() => {
    onAuthStateChanged( FirebaseAuth, async (user) => {
      if(!user) return dispatch( logout(null) );
      dispatch( login(user) );
    })
  }, [])
  
  if( status === 'checking' ) {
    return <CheckingAuth />
  }

  return (
    <BrowserRouter>
      <Routes>
        {
          status === 'authenticated'
          ? <Route path='/*' element={ <JournalRoutes />} />            
          : <Route path='/auth/*' element={ <AuthRoutes />} />
        }
        <Route path='/*' element={ <Navigate to='auth/login' />} />  
      </Routes>
    </BrowserRouter>
  )
}
