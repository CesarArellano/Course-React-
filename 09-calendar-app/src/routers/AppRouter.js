import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";

import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';

  // exact /login => LoginScreen
  // exact /  => Calendar
export const AppRouter = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/login" component={ LoginScreen }/>
        <Route exact path="/" component={ CalendarScreen }/>
        <Redirect to="/" />
      </Switch>
    </Router>
  )
}
