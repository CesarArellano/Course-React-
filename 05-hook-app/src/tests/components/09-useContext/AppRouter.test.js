import { shallow, mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Testing with AppRouter', () => {
  const user = {
    id: 1,
    name: 'CÃ©sar'
  };

  const wrapper = mount(
    <UserContext.Provider value={{
      user
    }}>
      <AppRouter />
    </UserContext.Provider>
  );
  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
})
