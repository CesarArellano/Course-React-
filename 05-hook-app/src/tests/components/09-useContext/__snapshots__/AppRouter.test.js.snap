// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with AppRouter should show correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "CÃ©sar"
}
      </pre>
    </div>
  </div>
</div>
`;
